### DISCUSSION TITLE
spacy-project-viz: visualize your spaCy project.yml files

### DISCUSSION THREAD

**Discussion by pmbaumgartner** [explosion/spaCy#7467] [Show & Tell]
# spacy-project-viz

Some of my `project.yml` files were getting a little complex to grok easily, so I built a CLI tool that will generate a graph visualization of commands, dependencies, outputs, and optionally variables in the project.yml file using [mermaid.js](https://mermaid-js.github.io/mermaid/#/) or [graphviz](https://graphviz.org/).

Repo: https://github.com/pmbaumgartner/spacy-project-viz

Example output (using the [pipelines/ner_demo_update](https://github.com/explosion/projects/blob/v3/pipelines/ner_demo_update/project.yml) project.yml) 

![spaCy project diagram](https://raw.githubusercontent.com/pmbaumgartner/spacy-project-viz/main/example/graphviz.svg)

I'd love to have folks play around with this and make any issues/PRs with any bugs or feature requests!

**Comment by svlandeg** [comment]
Ah, this is pretty cool! I'll definitely play with it, I already have a use-case in mind 😀
